tests/cpp/cell/test_single_wmma.cu

```diff
@@ -0,0 +1,200 @@
+#include "cell/mod.hpp"
+#include "common/test_utils.hpp"
+#include "cuda_utils.hpp"
+#include "util/debug.hpp"
+
+#include <glog/logging.h>
+
+namespace tiledcuda {
+
+using namespace cell;
+using namespace cell::copy;
+namespace tl = tile_layout;
+
+using namespace cute;
+
+namespace {
+
+// In this implementation, A and C are interpreted as being laid out in
+// row-major, and B is interpreted as being laid out in column-major.
+__device__ void naive_gemm(int kM, int kN, int kK,  //
+                           const __half* A, const __half* B, float* C) {
+    if (!thread0()) return;
+
+    for (int i = 0; i < kM; ++i) {
+        for (int j = 0; j < kN; ++j) {
+            float s = 0.;
+            for (int k = 0; k < kK; ++k) {
+                s += __half2float(A[i * kK + k]) * __half2float(B[k + kK * j]);
+            }
+            C[i * kN + j] = s;
+        }
+    }
+}
+
+__device__ void check_results(const float* hc1, const float* hc2, int numel) {
+    for (int i = 0; i < numel; ++i) assert(abs(hc1[i] - hc2[i]) < 1e-3);
+
+#if defined(DEBUG)
+    if (thread0()) {
+        printf("\n\nours:\n");
+        printf("%d:\t", 0);
+        for (int i = 0; i < numel; i++) {
+            printf("%.0f, ", hc1[i]);
+            if (i & (i + 1) % 32 == 0) printf("\n%d:\t", (i + 1) / 32);
+        }
+        printf("\nground-truth:\n");
+        printf("%d:\t", 0);
+        for (int i = 0; i < numel; i++) {
+            printf("%.0f, ", hc2[i]);
+            if (i & (i + 1) % 32 == 0) printf("\n%d:\t", (i + 1) / 32);
+        }
+    }
+#endif
+}
+
+template <typename Element, typename ElementAcc>
+__device__ void init_values(Element* a, Element* b, ElementAcc* c, int M, int N,
+                            int K) {
+    if (!thread0()) return;
+
+    for (int i = 0; i < M * K; ++i) a[i] = static_cast<Element>(i);
+    for (int i = 0; i < K * N; ++i) b[i] = static_cast<Element>(i);
+    for (int i = 0; i < M * N; ++i) c[i] = 0.;
+}
+
+template <typename Element, typename ElementAcc,  //
+          const int M, const int N, const int K,  //
+          typename TileIteratorA, typename RegA, typename LoadRegA,
+          typename TileIteratorB, typename RegB, typename LoadRegB,
+          typename SharedC, typename RegC, typename StoreRegC>
+__global__ void test_wmma(LoadRegA& load_rA, LoadRegB& load_rB,
```

Comment:
I wonder if the `load_rA`, `load_rB`, and `load_rC` have to be passed in via function pointers?

Reply:
Can we build a data structure similar to a `context` to store information related to generating the kernel? Too many templates seem a bit complex, but building a generic `context` also seems challenging...

Reply:
> Can we build a data structure similar to a `context` to store information related to generating the kernel? Too many templates seem a bit complex, but building a generic `context` also seems challenging...

Of course, I don't prefer too many template parameters. I took the most straightforward way in the current implementation.